server/service/transport_error.go

```diff
@@ -171,6 +171,17 @@ func encodeError(ctx context.Context, err error, w http.ResponseWriter) {
 		w.WriteHeader(http.StatusUnprocessableEntity)
 		enc.Encode(je)
 	default:
+		// when there's a tcp read timeout, the error is *net.OpError but the cause is an internal
+		// poll.DeadlineExceeded which we cannot match against, so we check the string here
```

Comment:
I think it would be best to keep the original error (before the call to `ctxerr.Cause` at the start of this function) and use `errors.As(oriErr, someNetOpErrorVar)` check. Based on your comment I think that should work?

Reply:
I had a similar thought but couldn't make up my mind.... now that you say it, I'll do that.